src/components/text-editor/prosemirror-adapter/plugins/image/inserter.ts

```
@@ -261,19 +283,253 @@
+ */
+const processDataUrlImage = (view: EditorView, dataUrl: string): void => {
+    // Extract mime type from data URL
+    const mimeMatch = /^data:([^;]+);/.exec(dataUrl);
+    const mimeType = mimeMatch ? mimeMatch[1] : 'image/png';
+    const extension = mimeType.split('/')[1] || 'png';
+
+    // Create a blob from the data URL
+    const regex = /^data:([^;]+);base64,(.+)$/;
+    const matches = regex.exec(dataUrl);
+    if (!matches) {
+        console.error('Invalid data URL format');
```

ESLint

[error] Unexpected console statement.

(no-console)